golang/cosmos/types/kv_entry.go

```diff
@@ -0,0 +1,111 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var _ json.Marshaler = &KVEntry{}
+var _ json.Unmarshaler = &KVEntry{}
+
+// KVEntry represents a string key / string value pair, where the value may be
+// missing. The semantics of a missing value are up to the user.
+type KVEntry struct {
+	key   string
```

Comment:
The key can be empty, however the semantics of that is usage dependent. I believe all current usages disallow an empty key. Maybe I will disallow empty keys since I can't imagine a correct usage for that.

Reply:
Yeah I remember why I didn't add that check. I didn't want to panic on construction of KVEntry with empty string key.